### ISSUE TITLE
Transition.slx model not found

### ISSUE THREAD

**Issue by valdezf10** [Big-Beef/DDPG_Drone#2]
Hi @M-Morris-95 ,

I am trying to run your SIMULINK model but it cannot find the "Transition.slx" model. There is a Transition.slxc file in there but it is not a SIMULINK model, it's a cache file. Did you forget to change the name to "Tricopter.slx"?  Cause that's how I was able to get it running. However, when running the training, there is no change in reward... it just stays at zero value. any ideas of why this is happening? 

Thanks.

**Comment by M-Morris-95** [comment]
Hello, 
Looks like something odd has happened there. I've uploaded what I think are the latest models, unfortunately it doesn't run very well on my laptop so it's difficult to verify it but looks like it's all there. I suggest looking at the dissertation file as that should have some explanations as to whats happening. Towards the end of the project the way I was saving files went a bit haywire but the dissertation should contain all the information on the equations and things I'm using. The appendix has some models in it which should be very similar to those on github

**Comment by valdezf10** [comment]
@M-Morris-95 ,
Thanks for the updated files. Also, the flight dynamics block doesn't seem to have anything as you explained in the paper (gravitational forces, propulsive forces and aerodynamic forces). When double clicking this block it seems like it's just a mass constant block... any ideas of what happened there?
How configurable would your Simulink model be if using a different aircraft model? Also, do you think MATLAB/Simulink is not the best platform for this type of work based on the results obtained and difficulties faced (as compared to Python/Pytorch/Tensorflow/etc world)? I am starting to do my research and am trying to decide which platform to use for DRL control of a fixed wing aircraft.  Thanks for sharing!

**Comment by M-Morris-95** [comment]
Hello, they should be there in the transition simulink model. I had a
similar problem to you and looked at using ROS/ardupilot/gazebo. Ultimately
matlab is best for physics and python is best for the RL, I tried combining
the 2 and it wasnt worth it. If your model is simple maybe try gazebo, I
think theres a built in plane and quad etc, if not then go with matlab for
sure

I can have another look in the morning but I've left uni now

Michael

**Comment by valdezf10** [comment]
I double checked but it doesn't seem to be there (see screenshot below).
![image](https://user-images.githubusercontent.com/44910796/68244303-659fed80-ffc9-11e9-96c4-0e55600189ff.png)

Thanks for the suggestions. I will give it a try in MATLAB and see how it goes! Another question, you have three .m agent files. I know that two correspond to the 'multirotor' model and the other to the 'transition' model. What about the agent.m? Do you have one only for the 'fixed wing' case (where you are not using the multirotor case)? Thank you.

I'm trying the 'multirotor' model and it seems like it's not converging as it was stated in the paper:
![image](https://user-images.githubusercontent.com/44910796/68247167-16f55200-ffcf-11e9-9110-2fd48faea7d5.png)
Maybe you didn't update the 'best' performed hyperparameters?

Thanks.
By the way, congrats on graduating!

**Comment by M-Morris-95** [comment]
I think that's the mask, if you click on the little arrow in the bottom
left of the box there should be more stuff there. The agent.m I'm not sure
what it is but probably some transition thing. You'll be able to set the
transition model to fixed wing by fixing the tilt angles and giving it an
initial forward velocity 10ms probably about right

**Comment by valdezf10** [comment]
@M-Morris-95 , Ahh, I miss that little arrow! Thanks for that.  Do you have any idea of why is not converging (as shown on the plot above)? I am trying to re-produce your results before I go on. Thanks.

**Comment by M-Morris-95** [comment]
I cant see the plot, but usually it was to do with the noise settings. A
good place to start is getting it to fly forwards in one direction with the
thrust angle of the motors fixed to vertical. Its quite a slow thing to
train as well so you might have to give it a few hours before you really
know. It took me weeks before I got it to converge properly.

At the moment I've only got my laptop which really cant run it fast enough
for me to play around with it but I'd suggest doing the very earliest
experiments in the report first. Limiting the rates and maximum attitude
will stop it going unstable which really kills training. Also adding in a
velocity controller in the multi rotor agent can be helpful, kinda defeats
the point of using RL but you'll gain intuition

Michael

**Comment by valdezf10** [comment]
Here is the screenshot:
![image](https://user-images.githubusercontent.com/44910796/68255640-a48e6d00-ffe2-11e9-85e9-ad35a3a89264.png)
I just ran the training with the parameters that you shared. I will give it a try with changing the noise settings and see the results. What type of GPU were you using for the training?
Also, I was wondering why you included PID controllers inside the Aircraft model while using DRL controller? Weren't you comparing PID vs DRL? What did you do to only run PID controllers for comparisons (without the DRL controller)? 

Sorry for asking too many questions.

Thanks for your help!

**Comment by M-Morris-95** [comment]
Hard to say why it's not converged there, if you look at the scopes in the
simulink model you can see what it's actually doing, sometimes it will have
a good solution even if the critic isnt perfect.

I used a gtx970, the performance increase was small as the majority of time
is spent on simulink not updating networks, my processor was overclocked to
4.5ghz which helped quite a bit. The things I did to speed it up are in the
report I think. If you have a lot of cores you can parallelise the whole
thing using matlab settings, its described in the bipedal robot example on
the mathworks site. Bare in mind that if you use all of your cores you wont
be able to use the computer while its running

The PID controllers are there for attitude and rate because they're already
optimised, a DRL agent could recreate them but it would be computationally
expensive for no real gain. The goal of the agent is higher level path
planning and getting the transition to work. If this had had better results
I would have used its solution for the transition on the physical aircraft

Michael

**Comment by M-Morris-95** [comment]
I've updated the github a bit more, put in a slightly longer readme but
more importantly added my backup models - while I was doing the project I
kept accidentally ruining the model and forgetting what I'd changed so
these were the fallback models which were definitely stable. Getting the
multirotor one to fly from A to B should be reasonably easy, as described
in 6.1 and 6.2 of the dissertation

**Comment by valdezf10** [comment]
Great, thank you so much for the info. It makes more sense now. I will try to see if I can get the Yaw control to work and see if I can reward engineer the reward function plus changing other things. This should be a good baseline for my project. I'll keep you updated!

Thank you again.
